version: "3.8"

services:
  contao:
    image: splintnet/dockerized-contao
    volumes:
      - app:/application
      - ./.env:/application/.env
      - ./files:/application/files
      - ./logs:/application/var/logs
      - assets:/application/assets
      - share:/application/share
    external_links:
      - db
    networks:
      - db
      - traefik

  nginx:
    image: nginx:alpine
    volumes:
      - ./docker/config/nginx.conf:/etc/nginx/conf.d/default.conf
      - app:/application
      - ./.env:/application/.env
      - ./files:/application/files
      - assets:/application/assets
      - share:/application/share
    depends_on:
      - contao
    networks:
      traefik:
    # labels:
      # - "traefik.enable=true"
      # - "traefik.http.routers.contao.rule=Host(`your-site.com`)"
      # - "traefik.http.routers.contao.tls.certresolver=myhttpchallenge"
      # - "traefik.http.routers.contao.entrypoints=websecure"
      # - "traefik.http.routers.contao.tls=true"

volumes:
  app:
  assets:
  share:

networks:
  db:
    external: true
  traefik:
    external: true
